xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -6985,6 +6986,19 @@ public boolean hasRecycleBin(XWikiContext context)
         return getStoreConfiguration().isRecycleBinEnabled();
     }
 
+    /**
+     * @see com.xpn.xwiki.api.XWiki#canSkipRecycleBin()
+     * @since 12.8RC1
+     */
+    public boolean canSkipRecycleBin()
```

Comment:
So in general the idea is to avoid putting more stuff in old-core especially new APIs. This sounds like a good API to go in refactorings module IMO. Moreover it would avoid using deprecated stuff such as Utils.getComponent.

Reply:
yep definitely not good in XWiki.java :) This seems like a configuration value so it must go in a *Configuration class for sure.

Reply:
We don't yet have a RefactoringConfiguration but that sounds like a better fit for this new configuration than XWiki indeed.

Reply:
Note: if we move it to refactoring then make sure to also change the configuration key which currently starts with "core".

Reply:
(And if it stays as a core config then it should go in CoreConfiguration)

Reply:
Thanks, I'll create `RefactoringConfiguration`.

Reply:
Done